components/modal/index.js

```diff
@@ -0,0 +1,203 @@
+		parentElement.removeChild( this.node );
+	}
+
+	/**
+	 * Opens or closes the modal based on whether the isOpen prop changed.
+	 *
+	 * @param {Object} prevProps The previous props.
+	 */
+	componentDidUpdate( prevProps ) {
+		const openStateChanged = this.props.isOpen !== prevProps.isOpen;
+		if ( openStateChanged && ! prevProps.isOpen ) {
+			this.openModal();
+		} else if ( openStateChanged && prevProps.isOpen ) {
+			this.closeModal();
+		}
+	}
+
+	/**
+	 * Closes the modal if it is open before unmount.
+	 */
+	componentWillUnmount() {
+		const { isOpen } = this.props;
+		if ( isOpen ) {
+			this.closeModal();
+		}
+		this.cleanDOM();
+	}
+
+	/**
+	 * Appends the modal's node to the DOM, so the portal can render the
+	 * modal in it. Also calls the openFirstModal when this is the first modal to be
+	 * opened.
+	 */
+	openModal() {
+		openModalCount++;
+
+		if ( openModalCount === 1 ) {
+			this.openFirstModal();
+		}
+	}
+
+	/**
+	 * Prepares the DOM for this modal and any additional modal to be mounted.
+	 *
+	 * It appends an additional div to the body for the modals to be rendered in,
+	 * it hides any other elements from screen-readers and adds an additional class
+	 * to the body to prevent scrolling while the modal is open.
+	 */
+	openFirstModal() {
+		ariaHelper.hideApp( parentElement );
+		document.body.classList.add( this.props.bodyOpenClassName );
+	}
+
+	/**
+	 * Removes the modal's node from the DOM. Also calls closeLastModal when this is
+	 * the last modal to be closed.
+	 */
+	closeModal() {
+		openModalCount--;
+
+		if ( openModalCount === 0 ) {
+			this.closeLastModal();
+		}
+	}
+
+	/**
+	 * Cleans up the DOM after the last modal is closed and makes the app available
+	 * for screen-readers again.
+	 */
+	closeLastModal() {
+		document.body.classList.remove( this.props.bodyOpenClassName );
+		ariaHelper.showApp();
+	}
+
+	/**
+	 * Renders the modal.
+	 *
+	 * @return {WPElement} The modal element.
+	 */
+	render() {
+		const {
+			isOpen,
+			overlayClassName,
+			className,
+			onRequestClose,
+			style: {
+				content,
+				overlay,
+			},
+			title,
+			icon,
+			closeButtonLabel,
+			children,
+			...otherProps
+		} = this.props;
+
+		if ( ! isOpen ) {
+			return null;
+		}
+
+		return createPortal(
+			<div
+				className={ classnames(
+					'components-modal__screen-overlay',
+					overlayClassName
+				) }
+				style={ overlay }>
+				<ModalFrame
+					style={ content }
+					className={ classnames(
+						'components-modal__frame',
+						className
+					) }
+					onRequestClose={ onRequestClose }
+					{ ...otherProps } >
+					<ModalHeader
+						closeLabel={ closeButtonLabel }
+						onClose={ onRequestClose }
+						title={ title }
+						icon={ icon } />
+					<div
+						className={ 'components-modal__content' }>
+						{ children }
+					</div>
+				</ModalFrame>
+			</div>,
+			this.node
+		);
+	}
+}
+
+Modal.defaultProps = {
+	bodyOpenClassName: 'modal-open',
+	role: 'dialog',
+	title: null,
```

Comment:
If a title prop is not passed, an empty H1 will be rendered, that should be definitely avoided:
`<h1 class="components-modal__header-heading"></h1>`

Also, if a title is passed the aria.labelledby should default to the ID value set on the title. Thus, the title should have an unique ID. Modals need to be labeled in any case, see comment in the thread for more details.

Reply:
>If a title prop is not passed, an empty H1 will be rendered, that should be definitely avoided:
`<h1 class="components-modal__header-heading"></h1>`

This comment has not been addressed.